hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/Record.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.hbtop;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+import javax.validation.constraints.NotNull;
+import org.apache.hadoop.hbase.hbtop.field.Field;
+import org.apache.hadoop.hbase.hbtop.field.FieldValue;
+import org.apache.yetus.audience.InterfaceAudience;
+
+
+/**
+ * Represents a record of the metrics in the top screen.
+ */
+@InterfaceAudience.Private
+public class Record implements Map<Field, FieldValue> {
+
+  private Map<Field, FieldValue> values;
+
+  public static final class Entry implements Map.Entry<Field, FieldValue> {
+    private final Field key;
+    private FieldValue value;
```

Comment:
This is settable making this object mutable. Would it be better if objects of this class were immutable? Just create a new instance when value changes? Or will that make it more complicated because this Entity is in a chain of Entities?

Reply:
Otherwise, will this be read by multiple threads? I suppose it not important if thread reads an old value because it doesn't notice change made by another thread?

Reply:
Will make this object immutable, thanks.

Reply:
I made Record class immutable in the latest commit. Can you please review it? @saintstack @joshelser